### ISSUE TITLE
LSP completion only works directly after -> or ::

### ISSUE THREAD

**Issue by jorams** [vimeo/psalm#2318]
Right now, LSP completions only work up to two characters away from the "gap", which is `->` or `::`.

```php
class A { function test() {} }
$a = new A;
// Typing 'te' completes 'test'
$a->te{st}
// Typing 'tes' completes nothing
$a->tes
```

I imagine VS Code just keeps using the previous completions if you keep typing, but not all clients do and even if you start later in an existing symbol there should still be completions.

A solution I have now (18454a5f6fc7ad8318575c9cb4f9035c1ea58532) is to check crudely whether or not we are still in the first name after the gap, and return completions if so. This fixes the above issue, but also produces quite a lot of false positives.

What could be a better way to approach this?

**Comment by muglug** [comment]
> but also produces quite a lot of false positives

What sort of false-positives?

**Comment by orklah** [comment]
Closing as stale
